File: chap14/fig14-7-left.R
library(dplyr)
library(ggplot2)

d <- read.csv('input/data-oil.csv')
N <- nrow(d)
K <- 2
fit_MAP <- readRDS('output/result-model14-5.RDS')
x_est <- fit_MAP$draws('x', format='matrix') %>% 
  matrix(., N, K)

d_plot <- data.frame(x_est, Class=factor(d$Class))
p <- ggplot(data=d_plot, aes(x=X1, y=X2, shape=Class)) +
  theme_bw(base_size=18) +
  geom_point(size=2.5, alpha=0.7) +
  scale_shape_manual(values=c(1,4,17))
ggsave(p, file='output/fig14-7-left.png', dpi=300, w=6, h=5)

File: chap14/fig14-7-left.py
import pandas
import cmdstanpy
from plotnine import ggplot, aes, theme_bw, geom_point, scale_shape_manual

d = pandas.read_csv('input/data-oil.csv')
N = len(d)
K = 2

fit_MAP = cmdstanpy.from_csv('output/result-model14-5')
x_est = fit_MAP.stan_variable('x')

d_plot = pandas.DataFrame(x_est, columns=['X1', 'X2'])
d_plot['Class'] = pandas.Categorical(d.Class, categories=[1,2,3])

p = (ggplot(d_plot, aes('X1', 'X2', shape='Class'))
    + theme_bw(base_size=18)
    + geom_point(size=2.5, alpha=0.7)
    + scale_shape_manual(values=['.', 'x', '^'])
)
p.save(filename='output/fig14-7-left.py.png', dpi=300, width=6, height=5)
